api/src/certification/enrolment/infrastructure/repositories/candidate-repository.js

```diff
@@ -70,12 +71,28 @@ export async function update(candidate) {
   }
 
   await knexConn('certification-subscriptions').where({ certificationCandidateId: candidate.id }).del();
+
   for (const subscription of candidate.subscriptions) {
-    await knexConn('certification-subscriptions').insert({
-      certificationCandidateId: candidate.id,
-      type: subscription.type,
-      complementaryCertificationId: subscription.complementaryCertificationId,
-    });
+    if (subscription.type === SUBSCRIPTION_TYPES.CORE) {
```

Comment:
Je prefererai me reposer sur le code metier du model 

```suggestion
    if (subscription.isCore()) {
```